BayesOpt/optimizer/mies.py

```diff
@@ -37,6 +38,7 @@ def __init__(self, search_space, obj_func, eq_func=None, ineq_func=None, x0=None
         self.ftarget = ftarget
         self.elitism = False
         self._penalty_func = dynamic_penalty
+        self._eval_type = eval_type
```

Comment:
Since mies is calling to the penalty function (outside BO library), I think it is necessary to maintain consistency with BO. Thus, the user's code will always deal with the same representation (list or dict).

Reply:
Fully agree :)